packages/common/src/hooks/purchaseContent/utils.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  UserTrackMetadata,
+  isPremiumContentUSDCPurchaseGated
+} from 'models/Track'
+
+import { payExtraAmountPresetValues } from './constants'
+import { PayExtraPreset, PurchasableTrackMetadata } from './types'
+
+export const getExtraAmount = (
+  amountPreset: PayExtraPreset,
+  customAmount = 0
+) => {
+  let extraAmount = 0
+  switch (amountPreset) {
+    case PayExtraPreset.LOW:
+    case PayExtraPreset.MEDIUM:
+    case PayExtraPreset.HIGH:
+      extraAmount = payExtraAmountPresetValues[amountPreset]
+      break
+    case PayExtraPreset.CUSTOM:
+      extraAmount = Number.isFinite(customAmount) ? customAmount : 0
```

Comment:
The input component itself doesn't allow negative numbers, so we're safe here. There is also form validation on the value that won't let us submit if we're outside the allowed range.
Though this could probably just be `Number.isNumber` instead.

Reply:
Oops, `Number.isNumber` is not a thing. The function used here is what we want. It requires that the argument be a number and not +/- infinity or NaN. We will rely on the form validation to handle the range.